### PR TITLE
integrate tts with azure-cognitiveservices-speech

### PR DESCRIPTION
**Pull request by albertofwb**

### CODE REVIEW

File: bot/config.py [hunk 1]
```diff
@@ -17,14 +17,20 @@
 use_chatgpt_api = config_yaml.get("use_chatgpt_api", True)
 allowed_telegram_usernames = config_yaml["allowed_telegram_usernames"]
 new_dialog_timeout = config_yaml["new_dialog_timeout"]
+mongo_host = config_env.get("MONGODB_HOST", "mongo")
 enable_message_streaming = config_yaml.get("enable_message_streaming", True)
-mongodb_uri = f"mongodb://mongo:{config_env['MONGODB_PORT']}"
+mongodb_uri = f"mongodb://{mongo_host}:{config_env['MONGODB_PORT']}"
```
Comment (clemsau) [inline bot/config.py:22]: What does allowing to change the mongo host has to do with allowing azure tts ?

File: bot/tts_utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+def get_tts_speak_audio_stream(text: str) -> bytes:
+    SPEECH_KEY = azure_tts_key
+    SPEECH_REGION = azure_tts_region
+    speech_config = speechsdk.SpeechConfig(subscription=SPEECH_KEY, region=SPEECH_REGION)
+    audio_config = speechsdk.audio.AudioOutputConfig(filename="audio.wav")
+
+    # The language of the voice that speaks.
+    speech_config.speech_synthesis_voice_name = azure_tts_name
+    speech_synthesizer = speechsdk.SpeechSynthesizer(speech_config=speech_config, audio_config=audio_config)
+    print("get text: {}".format(text))
```
Comment (clemsau) [inline bot/tts_utils.py:15]: Unwanted print

File: config/config.example.yml [hunk 3]
```diff
@@ -8,4 +8,10 @@ enable_message_streaming: true  # if set, messages will be shown to user word-by
...
 gpt_price_per_1000_tokens: 0.02
-whisper_price_per_1_min: 0.006
\ No newline at end of file
+whisper_price_per_1_min: 0.006
+
+
+enable_azure_tts: true
+azure_tts_key: ""
+azure_tts_region: ""
+azure_tts_voice: "zh-CN-XiaoxuanNeural"
```
Comment (clemsau) [inline config/config.example.yml:17]: Default should be english

**Comment by clemsau** [comment]
It could be an interesting idea, but IMHO integrating another 3rd party is not welcome as it makes the app more unstable, and increase the complexity only for edge cases.

File: config/config.example.yml (hunk 3 above)
Comment (albertofwb) [reply config/config.example.yml:17]: fine

File: bot/config.py [hunk 4]
```diff
@@ -17,13 +17,20 @@
...
 
 # chat_modes
-with open(config_dir / "chat_modes.yml", 'r') as f:
+with open(config_dir / "chat_modes.yml", 'r', encoding="utf8") as f:
```
Comment (albertofwb) [inline bot/config.py:25]: without define the encoding it faileds load chat_modes.yml on windows
because windows default encoding is unicode-16

**Comment by albertofwb** [comment]
> It could be an interesting idea, but IMHO integrating another 3rd party is not welcome as it makes the app more unstable, and increase the complexity only for edge cases.


the default settings for mstts is disabled.

**Pull request closed without merging**